tests/conftest.py

```diff
@@ -28,9 +28,47 @@
 import atexit
 import numpy as np
 from matlab.engine.matlabengine import MatlabFunc
-
-
-class MFWrapper:
+from abc import ABC, abstractmethod
+
+# Specification of a `TestEngine`
+class TestEngine(ABC):
+    @abstractmethod
+    def _repr_pretty(self, p):
+        """This placeholder can stay as is; it enables Hypothesis to work with this
+        runner as a fixture"""
+        return "Test Engine"
+
+    @abstractmethod
+    def convert(self, x):
+        """Convert the input to a type compatible with this engine. Can just be identity
+        if the runner is Python"""
+        return x
+
+    @abstractmethod
+    def equal(self, x, y) -> bool:
+        """Compare two values for equality in the representation used by this engine"""
+        pass
+
+# Python TestEngine
+class PythonEngine(TestEngine):
+    def _repr_pretty(sel, p):
+        return "Python Test Engine"
+
+    def convert(self, x):
+        return x
+
+    def equal(self, x, y) -> bool:
+        return x == y
+
+    # Overload calling of methods and 'rethrow' to global context
+    def __getattribute__(self,name):
+        def newfunc(*args, **kwargs):
+          result = eval(name + "(*" + str(args) + ", *" + str(kwargs) + ")")
+          return result
+        return newfunc
```

Comment:
```suggestion
def __getattribute__(self, name):
    def newfunc(*args, **kwargs):
        func = globals().get(name)  # Access global dictionary of defined functions
        if callable(func):
            return func(*args, **kwargs)
        raise AttributeError(f"'{name}' is not callable")
    return newfunc
```
This way can avoid the use of eval.